src/functional/fencil_processors/callables/cpp/build.py

```diff
@@ -0,0 +1,191 @@
+import pathlib
+import subprocess
+import textwrap
+from typing import Dict, Final, Optional, Sequence
+
+from eve import Node
+from eve.codegen import JinjaTemplate, TemplatedGenerator
+from functional.fencil_processors import defs
+
+
+_build_subdir: Final = "build"
+
+
+class FindDependency(Node):
+    name: str
+    version: str
+
+
+class LinkDependency(Node):
+    name: str
+    target: str
+
+
+class CMakeListsFile(Node):
+    project_name: str
+    find_deps: Sequence[FindDependency]
+    link_deps: Sequence[LinkDependency]
+    source_names: Sequence[str]
+    bin_output_suffix: str
+
+
+class CMakeListsGenerator(TemplatedGenerator):
+    CMakeListsFile = JinjaTemplate(
+        """
+        project({{project_name}})
+        cmake_minimum_required(VERSION 3.20.0)
+
+        # Languages
+        enable_language(CXX)
+        set(CMAKE_CXX_STANDARD 17)
+        set(THREADS_PREFER_PTHREAD_FLAG ON)
+        find_package(Threads REQUIRED)
+        link_libraries(Threads::Threads)
```

Comment:
The code of the bindings requires C++17 regardless of what GridTools requires, so that should be specified anyways. I moved it to `target_compile_features` to make it clearer and in line with modern CMake. 

I usually link to pthreads by default because otherwise you can't use threading from the standard library. The bindings don't currently use threading, so I removed it.